### PR TITLE
feat: add lp fee pct to API

### PR DESCRIPTION
**Pull request by mrice32**

### CODE REVIEW

File: api/suggested-fees.js [hunk 1]
```diff
@@ -36,32 +60,41 @@ const handler = async (request, response) => {
   }
 };
 
-
 const getTokenDetails = async (provider, l2Token, chainId) => {
-  const bridgeAdmin = BridgeAdminEthers__factory.connect("0x30B44C676A05F1264d1dE9cC31dB5F2A945186b6", provider);
+  const bridgeAdmin = BridgeAdminEthers__factory.connect(
```
Comment (mrice32) [inline api/suggested-fees.js:64]: Note: a lot of these changes are just formatting.

File: api/suggested-fees.js [hunk 2]
```diff
@@ -1,27 +1,51 @@
...
+
+    const [depositFeeDetails, lpFee] = await Promise.all([
+      sdk.across.gasFeeCalculator.getDepositFeesDetails(
+        provider,
+        amount,
+        l1Token === sdk.across.constants.ADDRESSES.WETH
+          ? sdk.across.constants.ADDRESSES.ETH
+          : l1Token
+      ),
+      lpFeeCalculator.getLpFeePct(l1Token, bridgePool, amount, parsedTimestamp),
```
Comment (mrice32) [inline api/suggested-fees.js:39]: This is new.

File: api/suggested-fees.js [hunk 3]
```diff
@@ -1,27 +1,51 @@
...
+    const provider = new ethers.providers.StaticJsonRpcProvider(
+      `https://mainnet.infura.io/v3/${REACT_APP_PUBLIC_INFURA_ID}`
+    );
+
+    let { amount, l2Token, chainId, timestamp } = request.query;
+    if (!isString(amount) || !isString(l2Token) || !isString(chainId))
+      throw new InputError("Must provide amount and token as query params");
+    const parsedTimestamp = isString(timestamp)
+      ? Number(timestamp)
+      : (await provider.getBlock("latest")).timestamp;
```
Comment (mrice32) [inline api/suggested-fees.js:20]: This is also new.

File: src/components/CoinSelection/CoinSelection.styles.ts [hunk 4]
```diff
@@ -11,7 +11,9 @@ export const Wrapper = styled.div`
   flex-direction: column;
 `;
 // User agents wars, so have to check for chrome too
-const isSafari = navigator.userAgent.search("Safari") >= 0 && navigator.userAgent.search("Chrome") < 0
+const isSafari =
```
Comment (mrice32) [inline src/components/CoinSelection/CoinSelection.styles.ts:14]: Just formatting.

**Review by Gamaranto — APPROVED** [review]
just a question on my side, otherwise looks good

**Review by daywiss — APPROVED** [review]
lgtm, i did have some question about what was being returned in `getTokenDetails` but looked up the whitelist event and it made sense

File: api/suggested-fees.js [hunk 5]
```diff
@@ -1,27 +1,51 @@
...
-    let { l1Token } = await getTokenDetails(provider, l2Token, chainId);
-    if (l1Token === sdk.across.constants.ADDRESSES.WETH) l1Token = sdk.across.constants.ADDRESSES.WETH;
-    const depositFeeDetails = await sdk.across.gasFeeCalculator.getDepositFeesDetails(provider, amount, l1Token === sdk.across.constants.ADDRESSES.WETH ? sdk.across.constants.ADDRESSES.ETH : l1Token);
-    if (depositFeeDetails.isAmountTooLow) throw new InputError("Sent amount is too low relative to fees");
-    
     const responseJson = {
       slowFeePct: depositFeeDetails.slow.pct,
-      instantFeePct: depositFeeDetails.instant.pct
+      instantFeePct: depositFeeDetails.instant.pct,
+      lpFeePct: lpFee.toString(),
```
[the remark this answers is not in the document]
Comment (mrice32) [reply api/suggested-fees.js:49]: Yep, they are all strings. I'm not sure if JSON.stringify that's likely being called under the hood would be handling BigNumbers appropriately, so I think strings are safer.

**Pull request merged**